eng/tools/typespec-validation/src/rules/sdk-tspconfig-validation.ts

```diff
@@ -229,24 +257,38 @@ export class TspConfigTsDpPackageDirectorySubRule extends TspconfigEmitterOption
   constructor() {
     super("@azure-tools/typespec-ts", "package-dir", new RegExp(/^(?:[a-z]+-)*rest$/));
   }
-  protected skip(_: any, folder: string) {
-    return skipForManagementPlane(folder);
+  protected skip(config: any, folder: string) {
+    return skipForModularOrManagementPlaneInTsEmitter(config, folder);
   }
 }
 
-export class TspConfigTsDpPackageNameMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
+export class TspConfigTsRlcDpPackageNameMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
   constructor() {
     super(
       "@azure-tools/typespec-ts",
       "package-details.name",
       new RegExp(/^\@azure-rest\/[a-z]+(?:-[a-z]+)*$/),
     );
   }
-  protected skip(_: any, folder: string) {
-    return skipForManagementPlane(folder);
+
+  protected skip(config: any, folder: string) {
+    return skipForModularOrManagementPlaneInTsEmitter(config, folder);
   }
 }
 
+export class TspConfigTsMlcDpPackageNameMatchPatternSubRule extends TspconfigEmitterOptionsSubRuleBase {
+  constructor() {
+    super(
+      "@azure-tools/typespec-ts",
+      "package-details.name",
+      new RegExp(/^\@azure\/[a-z]+(?:-[a-z]+)*$/),
```

Comment:
```suggestion
      new RegExp(/^@azure\/(?:[a-z]+-)*[a-z]+$/),
```
Please keep the same to regex described the issue